source/common/http/http2/codec_impl.cc

```diff
@@ -1265,6 +1380,14 @@ int ConnectionImpl::onStreamClose(int32_t stream_id, uint32_t error_code) {
       }
 
       stream->runResetCallbacks(reason);
+    } else if (!stream->reset_reason_.has_value() && stream->stream_manager_.hasBufferedData()) {
```

Comment:
Could you explain the sequence of events that can lead you to get here?  It's unclear to me why this appears relevant in cases where defer_processing_backedup_streams_ is false.